### PR TITLE
Use different colormaps for subsequent segs when combined into 1 FSLeyes command

### LINKED ISSUE #4693: Unify FSLeyes commands for multiclass DeepSeg models
FSLeyes commands outputted by `sct_deepseg -i t2.nii.gz -task seg_sc_lesion_t2w_sci` differ from the FSLeyes command in [our tutorial](https://github.com/spinalcordtoolbox/sct_tutorial_data/blob/8b91e72c6575cbf40c5a66898fe3a13d9c26790c/single_subject/batch_single_subject.sh#L128):

Screenshot 2024-11-12 at 11 16 44 AM

We could consider unifying the commands. I personally like the command from our tutorial, which opens both spinal cord and lesion segmentation in a single FSLeyes window:

```console
fsleyes t2.nii.gz -cm greyscale t2_sc_seg.nii.gz -cm red -a 70.0 t2_lesion_seg.nii.gz -cm blue-lightblue -a 70.0 &
```

### PR DESCRIPTION
**Pull request by joshuacwnewton**
## Description

For the following command:

```
sct_deepseg -task seg_sc_lesion_t2w_sci -i t2_fake_lesion.nii.gz
```

Two segs are output (SC seg, lesion seg). Comparing the FSLeyes commands before and after this PR...

Before (both red):

image

After (one red, one blue):

image

## Linked issues

Fixes #4693.

### CODE REVIEW

File: spinalcordtoolbox/utils/shell.py [hunk 1]
```diff
@@ -154,7 +171,11 @@ def _construct_itksnap_syntax(viewer, files, im_types):
         if not im_types:
             gray_images.append(files[i])
         else:
-            colormap = IMTYPES_COLORMAP[im_types[i]]['itksnap']
+            # itksnap only has 1 colormap per seg, so always use cm "1" for segs
```
Comment (mguaypaq) [inline spinalcordtoolbox/utils/shell.py:174]: Thanks for this comment.

**Review by mguaypaq — APPROVED** [review]
Very clean, thanks!

File: spinalcordtoolbox/utils/shell.py [hunk 2]
```diff
@@ -109,14 +114,20 @@ def display_viewer_syntax(files, verbose, im_types=[], minmax=[], opacities=[],
 
 def _construct_fslview_syntax(viewer, files, im_types, minmax, opacities, mode):
     cmd = viewer
+    n = 0
```
[the remark this answers is not in the document]
Comment (joshuacwnewton) [reply spinalcordtoolbox/utils/shell.py:117]: Exactly... I initially had them use separate counters, but then I realized overlaying a red seg on top of a red softseg probably isn't what we want (if we were to mix seg and softseg)?

**Comment by valosekj** [comment]
Thank you for the fix, Joshua!

**Pull request merged**